### ISSUE TITLE
RGB setting,way more worse Rendering? Wrong Color blocks increase?

### ISSUE THREAD

**Issue by SlamMate** [muskie82/MonoGS#94]
I ran the eval mode in the A6000 on fr3/office without any modifications. But something went wrong there. In the initial few frames, it looks okay(but still not okay as the paper presents), but later as the drift continues the rendering becomes unacceptable worse. 
In the beginning part, the colour block looks fine. (first is rendering image, second is input image)
![image_1](https://github.com/muskie82/MonoGS/assets/96341196/b67d73d5-9077-4308-a1fe-8098455db906)
![image_2](https://github.com/muskie82/MonoGS/assets/96341196/52317aea-71ac-46d8-b594-fc3d649297a8)

It is getting worse.
![image_1](https://github.com/muskie82/MonoGS/assets/96341196/b3eac8dd-85b8-4a32-886d-570b2bc21926)
![image_2](https://github.com/muskie82/MonoGS/assets/96341196/f105fc16-2299-4bde-8098-56415b12b3f5)

In the end, it's very blur, Like a piece of art.
![image_1_saved_images](https://github.com/muskie82/MonoGS/assets/96341196/f1aff737-eeff-47ca-b24a-f3a255a4fb2c)
![image_2_saved_images](https://github.com/muskie82/MonoGS/assets/96341196/721a3509-5914-4c9b-b90d-3b9f2552bd89)

The ATE is the same as report, 0.04m.
![Screenshot from 2024-05-21 16-11-24](https://github.com/muskie82/MonoGS/assets/96341196/d7ed8014-c00a-4a08-8a5d-8890185cd8e4)

Why the rendering image is so awful...Couldn't even track in the end? Thank you for your patience~
The traj is okay too:
![image](https://github.com/muskie82/MonoGS/assets/96341196/94f32f3f-6598-4551-9685-1dc8d3d0d04d)

Below is my visualization code:
`def save_images(image, folder="InputS"):

    if len(image.shape) == 3:
        image = image.unsqueeze(0)
    image = image.cpu()
    image = image.detach().numpy()
    for i, img in enumerate(image):
        img = np.transpose(img, (1, 2, 0))
        
        img = (img * 255).astype(np.uint8)
        
        img_pil = Image.fromarray(img)
        img_pil.save(os.path.join(folder, f"image_{i+1}.png"))`
input's shape is (3, 480, 640)

**Comment by SlamMate** [comment]
I found some relevant issues in both MonoGS and 3DGS. 

1. https://github.com/muskie82/MonoGS/issues/60
2. https://github.com/graphdeco-inria/gaussian-splatting/issues/682
3. https://github.com/graphdeco-inria/gaussian-splatting/issues/424
4. https://github.com/graphdeco-inria/gaussian-splatting/issues/770

I think these rendering errors have to do with the sensitivity of the Gaussian reconstruction itself to camera distance and the novel view. I've read a lot of issues saying that the consistency of 3DGS is not very good. And the good tracking results are due to the fact that the camera position is adjusted according to the gradient of the loss function, which makes the rendered view better when it finally converges?

**Comment by SlamMate** [comment]
When I add more visualizations in the front end, The ATE become worse, even 1.07m. The picture becomes worse. I mean no one uses visualization in MonoGS?

**Comment by HerrSchnapps** [comment]
> When I add more visualizations in the front end, The ATE become worse, even 1.07m. The picture becomes worse. I mean no one uses visualization in MonoGS?

DId you solve this problem? I found it as well. After I try to speed up the rendering function, the quality of results is not good.
